zimports/zimports.py

```diff
@@ -172,29 +197,79 @@ def rewrite(self):
 
 
 class TypeCheckingBlocks:
-    def __init__(self, source_lines):
+    def __init__(self, source_lines, type):
         self.type_checking_blocks = []
+        self.anti_type_checking_blocks = []
+        self.type = type
+
         in_type_checking_block = False
+        in_anti_type_checking_block = False
+
         for lineno, line in enumerate(source_lines, 1):
             if re.match(r"^if [\w+\.]*TYPE_CHECKING:", line):
                 # we are inside an "if TYPE_CHECKING:" block
                 in_type_checking_block = True
-                self.type_checking_blocks.append((lineno, set()))
+                self.type_checking_blocks.append((lineno, set(), -1))
             elif in_type_checking_block:
-                if line and not re.match(r"^\s+(from|import)", line):
-                    # line is not empty and not an indented import so the
-                    # type checking import block has ended
-                    in_type_checking_block = False
+                if line and re.match(r"^(?:else|elif)", line):
+                    # it's an else.  new typing block
+                    in_anti_type_checking_block = True
+                    self.anti_type_checking_blocks.append(
+                        (lineno, set(), self.type_checking_blocks[-1][0])
+                    )
+                elif line and re.match(r"^[a-zA-Z]", line):
```

Comment:
number is also legal here. not very useful but technically not a syntax error:

```py
if TYPE_CHECKING:
   ...
123
```

Reply:
really?  hm .   well tool works for us :)

Reply:
bzzt.  can also be an underscore.   see #34

Reply:
I guess it could also start with `-` in a negative number. But that's hardly common

Reply:
also `~` is a legal char!